controller.h

```diff
@@ -380,45 +389,57 @@ class Controller : public IController
 
     void Stop() override
     {
+        std::lock_guard<std::mutex> lock(_canvasMutex);
         logger->debug("Stopping canvases...");
 
         for (auto &canvas : _canvases)
             canvas->Effects().Stop();
     }
 
-    bool AddCanvas(unique_ptr<ICanvas> ptrCanvas) override
+    uint32_t AddCanvas(unique_ptr<ICanvas> ptrCanvas) override
     {
         logger->debug("Adding canvas {}...", ptrCanvas->Name());
 
         // This is a bit odd; we try get the current canvas with the ID specified by the new one,
         // and we only proceed in the exception case if the canvas doesn't exist, where we add it
 
+        std::lock_guard<std::mutex> lock(_canvasMutex);
         try
         {
             GetCanvasById(ptrCanvas->Id());
             logger->error("Canvas with ID {} already exists.", ptrCanvas->Id());
-            return false;
+            return -1;
```

Comment:
I'm not stoked about this in-band signal... it feels like std::expected is that way to do it "right" but that's C++23 and I don't think we have that under Ubuntu yet.  So I blame Ubuntu.